[docs] Docs for plugins and small issues (#23179)
MystenLabs/sui: docs/content/references/contribute/mdx-components.mdx

import YTCarousel from "@site/src/components/YTCarousel"

<YTCarousel ids={["dQw4w9WgXcQ","EwTZ2xpQwpA","1TewCPi92ro"]}/>

## Step titles

> When creating process documents it's best to use an ordered list with each step consisting of a single instruction for the user. On occasion, however, you might want to have a group of instructions for each step. The step titles enable the use of numbered steps without having to maintain the proper numbered step order. 

Reviewer: ```suggestion
It is recommended to use an ordered list for each step in a process document consisting of a single instruction for the user. On occasion, however, you might want to have a group of instructions for each step. The step titles enable the use of numbered sub-steps.
```

### When to use

> Use when you detail a long process that deals with multiple technologies or otherwise requires lengthy contextual content. You don't, for instance, want to create a 30-step ordered list or one where list item 1 is three full page scrolls to get to list item 2. Having steps in the title of longer processes helps users recall where they left off and more easily navigate back using the ToC if they don't complete the task set.

Reviewer: ```suggestion
Use when you detail a long process with multiple technologies, or a process that otherwise requires lengthy contextual content. Do not create a verbose ordered list or one where list item #1 requires several pages of depth before the reader gets to list item #2. Having steps in the title of longer processes helps readers recall where they left off if they don't complete the task set, and enables them to easily navigate back using the ToC. 
```

### How to use

To create step headings, add a `step` or `substep` suffix to the usual heading levels. 

> For instance, to create a level 2 heading, you'd write `##step Buy peanuts`, which is the same as typing `## Step 1: Buy peanuts`.

Reviewer: ```suggestion
For instance, to create a level 2 heading, write `##step Buy peanuts`, which is the same as typing `## Step 1: Buy peanuts`.
```

> The feature supports one level of [`nesting`](/) using the `substep` suffix in the same manner. Don't forget to use the proper number of hashes in relation to the parent step. Continuing the example, you might have `###substep Buy Cracker Jacks`, which would be the same as typing `### Step 1.1: Buy Cracker Jacks`.

Reviewer: ⚾
